### ISSUE TITLE
application manifest does not correctly capture some applications' URLs

### ISSUE THREAD

**Issue by glyn** [cloudfoundry/cli#765]
If an application has multiple domains and hostnames, this isn't captured correctly when the application's manifest is created. In particular, another application created from that manifest ends up with a different collection of URLs.

### 1. Push an application without using a manifest
```
$ cf create-domain micropcf-org example.com
$ cf push app2 -b https://github.com/cloudfoundry/staticfile-buildpack.git -d example.com
$ cf create-domain micropcf-org example2.com
$ cf push app2 -b https://github.com/cloudfoundry/staticfile-buildpack.git -d example2.com
$ cf create-route micropcf-space example.com --hostname hostA
$ cf map-route app2 example.com --hostname hostA
```
### 2. Check the application's URLs
At this point, the application has *three* URLs (`hostA.example2.com` is *not* one of them):
```
$ cf apps
Getting apps in org micropcf-org / space micropcf-space as admin...
OK

name  ...   urls   
app2  ...   app2.example.com, app2.example2.com, hostA.example.com   
```
(Details elided.)
### 3. Get the application's manifest
```
$ cf create-app-manifest app2
$ cat app2_manifest.yml 
---
applications:
- name: app2
  memory: 1024M
  instances: 1
  buildpack: https://github.com/cloudfoundry/staticfile-buildpack.git
  hosts:
  - hostA
  - app2
  domains:
  - example.com
  - example2.com
```
### 4. Deploy another application using this manifest
```
$ cf push -f ./app2_manifest.yml 
Using manifest file ./app2_manifest.yml

Creating app app3 in org micropcf-org / space micropcf-space as admin...
OK

Using route hostA.example.com
Binding hostA.example.com to app3...
OK

Using route app2.example.com
Binding app2.example.com to app3...
OK

Creating route hostA.example2.com...
OK

Binding hostA.example2.com to app3...
OK

Using route app2.example2.com
Binding app2.example2.com to app3...
OK

Uploading app3...
... etc. ...
```
### 5. Check the appliction's URLs
This application has an extra URL:
```
$ cf apps
Getting apps in org micropcf-org / space micropcf-space as admin...
OK

name   ...   urls   
app2   ...   app2.example.com, app2.example2.com, hostA.example.com   
app3   ...   app2.example.com, app2.example2.com, hostA.example.com, hostA.example2.com
```
(Details elided.)

**Comment by dkoper** [comment]
Hi @glyn ,

Your reporting style is awesome!
The issue with the current app manifest specification is that host and domain fields are not paired, so the only way for create-app-manifest to express the multiple urls is the way you displayed, and push then creates the cross-product.
I think this will get worse as we add route paths and ports. I am therefore working on a proposal to define a new way of expressing routes in app manifests, based on a proposal from @shalako. I am looking forward to your feedback when I share it on CF Dev in the near future.

Cheers,
Dies Koper
CF CLI PM

**Comment by glyn** [comment]
@dkoper Fantastic. A bit of design would go a long way here. Thanks.

**Comment by glyn** [comment]
I should mention I am pairing with @benpatt, so must share the credit with him. ;-)

**Comment by dkoper** [comment]
This has now been fixed in cf CLI 6.21.0.
https://github.com/cloudfoundry/cli#downloads
